### ISSUE TITLE
Add route with query

### ISSUE THREAD

**Issue by tinu73** [dfleury2/beauty#8]
Is it possible to add a route with queries, e.g. server.add_route("/person?id") and then in the request http://127.0.0.1:8085/person?id=1

**Comment by dfleury2** [comment]
Hi,
in this case, the id is an attributes, not a part of the route.
You can get it using attributes if the request.

Here an example based on server.cpp:
```cpp
    server.add_route("/person")
        // Get the list of available ids
        .get([](const auto& req, auto& res) {

            auto id = req.a("id").as_string();
            if (id.empty()) {
                for (const auto& [id, _] : storage) {
                    res.body() += id + "\n";
                }
            }
            else {
                std::string response = "Id: " + id + " not found";
                if (auto found = storage.find(id); found != end(storage)) {
                    res.body() = "id: " + id + ", name: " + found->second + "\n";
                }
                else {
                    res.body() = "Id: " + id + " NOT FOUND";
                }
            }
        })
```

**Comment by tinu73** [comment]
Thank you  very much for you help. Issue can be closed.

**Comment by tinu73** [comment]
I thought its best to place this question right to the question above, since they are related.

I have a the following route:

`server.add_route("/channel/setChannel/:value")
            .get([&](const auto& req, auto& res) {
                auto channel = req.a("value").as_string();
                ...          
                res.body() = "ok";
            });`

GET Request with Postman: `localhost:8090/channel/setChannel/{"indoor":true,"outdoor":{"left":false,"right":true}}`. The GET request from Postman is url-encoded: `%7B%22indoor%22:true,%22outdoor%22:%7B%22left%22:false,%22right%22:false%7D%7D`.

`auto channel = req.a("value").as_string();` results in `12334indoor34:true,34outdoor34:12334left34:false,34right34:false125125`, but instead i would like to receive `{"indoor":true,"outdoor":{"left":false,"right":true}}`.

How to solve this issue? It must be a GET request, since the request is from external system. I cannot change to POST with body application/json.

**Comment by dfleury2** [comment]
Hi,
I just commited an example in examples/server.cpp to try to reproduce your problem.
using: 
`wget 'http://localhost:8085/channel/setChannel/{"indoor":true,"outdoor":{"left":false,"right":true}}'` 
I got :
`value: {"indoor":true,"outdoor":{"left":false,"right":true}}`

I do not use postman, so I do not know how to reproduce you issue. Did you use the last version of Beauty ?

**Comment by tinu73** [comment]
Hi,
i really appreciate your prompt feedback. Basically Postman is just a tool, but you are right that wget does the job as well, the url is encoded:

```
wget 'http://localhost:8090/channel/setChannel/{"indoor":true,"outdoor":{"left":false,"right":true}}'
--2024-01-25 14:35:26--  http://localhost:8090/channel/setChannel/%7B%22indoor%22:true,%22outdoor%22:%7B%22left%22:false,%22right%22:true%7D%7D
```
It looks like that i have to update Beauty to the last version, since i get the same result with wget:
`12334indoor34:true,34outdoor34:12334left34:false,34right34:true125125`

Using `curl --data "value={\"indoor\":false,\"outdoor\":{\"left\":true,\"right\":true}}" http://172.52.71.10:8090/channel/setChannel/ --get` i get the correct output:

`{"indoor":true,"outdoor":{"left":false,"right":true}}`

I will check latest version an let you know.

**Comment by tinu73** [comment]
problem solved, i used version 1.0.2. May thanks. Issue can be closed

**Comment by dfleury2** [comment]
Great.

I try to make tags stable, but sometimes I forget to make release on github (neither on conan)
